### PR TITLE
intelligently search for bower_components

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

**Comment by garlicnation** [comment]
@nomego This will resolve some sublimelinter problems.

File: bin/polylint.js [hunk 1]
```diff
@@ -170,6 +171,29 @@ process.on('unhandledRejection', function(reason, p) {
   fatalFailureOccurred = true;
 });
 
+// Find the bower dir.
+var searchPath = options.root;
```
Comment (rictic) [inline bin/polylint.js]: This is only used one place, probably easier to read if you inline it there

File: bin/polylint.js [hunk 2]
```diff
@@ -170,6 +171,29 @@ process.on('unhandledRejection', function(reason, p) {
...
+var parentDirs = [];
+var foundBower = false;
+while (!foundBower) {
+  var maybeBower = path.resolve.apply(undefined, [searchPath].concat(parentDirs).concat([options.bowerdir]));
```
Comment (rictic) [inline bin/polylint.js]: Consider renaming maybeBower to something a bit clearer, like candidatePath

File: bin/polylint.js [hunk 3]
```diff
@@ -170,6 +171,29 @@ process.on('unhandledRejection', function(reason, p) {
...
+    stat = fs.statSync(maybeBower);
+    foundBower = true;
+  } catch (err) {
+    parentDirs.push('..');
+  }
+}
+if (!foundBower) {
+  options.bowerdir = undefined;
+} else {
+  options.bowerdir = path.join(path.join(parentDirs), options.bowerdir);
```
Comment (rictic) [inline bin/polylint.js]: This is kinda unclear, you're using `options.bowerdir` to find `options.bowerdir`?

I guess it's initally a partial path, like 'paper-button/bower.json' and you're walking up the tree of parents?

Comment (rictic) [inline bin/polylint.js]: Would it be worth it to do path.resolve here?

Comment (garlicnation) [inline bin/polylint.js]: No, since fsResolver wants a path relative to `root`.

**Comment by rictic** [comment]
:+1:

**Pull request merged**